src/backend/distributed/master/master_delete_protocol.c

```diff
@@ -324,14 +324,24 @@ DropShards(Oid relationId, char *schemaName, char *relationName,
 	ListCell *shardIntervalCell = NULL;
 	int droppedShardCount = 0;
 
+	if (XactModificationLevel != XACT_MODIFICATION_NONE)
+	{
+		ereport(ERROR, (errcode(ERRCODE_ACTIVE_SQL_TRANSACTION),
+						errmsg("shard drop operations must not appear in "
+							   "transaction blocks containing other distributed "
+							   "modifications")));
+	}
+
+	BeginOrContinueCoordinatedTransaction();
+	if (MultiShardCommitProtocol == COMMIT_PROTOCOL_2PC)
```

Comment:
That's not really an argument against using 2PC for them - the master could die inbetween, which'd be annoying to recover from.